service/app/utils/module/OnRequestHandler.java

```diff
@@ -82,4 +97,110 @@ public Action createAction(Http.Request request, Method method) {
   }
+
+  /**
+   * Set the Context paramter to the request
+   *
+   * @param httpReq
+   * @param userId
+   */
+  private void initializeContext(Http.Request httpReq, String userId) {
+    try {
+      Map<String, Object> requestContext = new WeakHashMap<>();
+      String env = getEnv(httpReq);
+      requestContext.put(JsonKey.ENV, env);
+      requestContext.put(JsonKey.REQUEST_TYPE, JsonKey.API_CALL);
+      Optional<String> optionalChannel = httpReq.getHeaders().get(HeaderParam.CHANNEL_ID.getName());
+      String channel;
+      if (optionalChannel.isPresent()) {
+        channel = optionalChannel.get();
+      } else {
+        channel = getCustodianOrgHashTagId();
+      }
+      requestContext.put(JsonKey.CHANNEL, channel);
+      requestContext.put(
+          JsonKey.REQUEST_ID, httpReq.getHeaders().get(JsonKey.REQUEST_MESSAGE_ID).get());
+      requestContext.putAll(cacheTelemetryPdata());
+      Optional<String> optionalAppId = httpReq.getHeaders().get(HeaderParam.X_APP_ID.getName());
+      if (optionalAppId.isPresent()) {
+        requestContext.put(JsonKey.APP_ID, optionalAppId.get());
+      }
+      Optional<String> optionalDeviceId =
+          httpReq.getHeaders().get(HeaderParam.X_Device_ID.getName());
+      if (optionalDeviceId.isPresent()) {
+        requestContext.put(JsonKey.DEVICE_ID, optionalDeviceId.get());
+      }
+      if (null != userId) {
+        requestContext.put(JsonKey.ACTOR_ID, userId);
+        requestContext.put(JsonKey.ACTOR_TYPE, StringUtils.capitalize(JsonKey.USER));
+      } else {
+        Optional<String> optionalConsumerId =
+            httpReq.getHeaders().get(HeaderParam.X_Consumer_ID.getName());
+        String consumerId;
+        if (optionalConsumerId.isPresent()) {
+          consumerId = optionalConsumerId.get();
+        } else {
+          consumerId = JsonKey.DEFAULT_CONSUMER_ID;
+        }
+        requestContext.put(JsonKey.ACTOR_ID, consumerId);
+        requestContext.put(JsonKey.ACTOR_TYPE, StringUtils.capitalize(JsonKey.CONSUMER));
+      }
+      Map<String, Object> map = new WeakHashMap<>();
+      map.put(JsonKey.CONTEXT, requestContext);
+      httpReq.flash().put(JsonKey.CONTEXT, mapper.writeValueAsString(map));
+    } catch (Exception ex) {
+      logger.error("Error process set request context" + ex.getMessage());
+      throw new BaseException(
+          IResponseMessage.SERVER_ERROR,
+          IResponseMessage.INTERNAL_ERROR,
+          ResponseCode.SERVER_ERROR.getCode());
+    }
+  }
+
+  private String getCustodianOrgHashTagId() {
+    if (null != custodianOrgHashTagId) {
+      return custodianOrgHashTagId;
+    }
+    synchronized (OnRequestHandler.class) {
+      if (custodianOrgHashTagId == null) {
+        try {
+          // Get hash tag ID of custodian org
+          String custodianOrgId = CacheUtil.getConfigSettings().get(JsonKey.CUSTODIAN_ORG_ID);
+          Response response = userService.getOrganisationDetails(custodianOrgId);
```

Comment:
Cust org id hash tag or other details do not change per api. Any reason, why this can't be cached as well?